pyproject.toml

```diff
@@ -22,3 +22,81 @@ environment = 'RUSTUP_TOOLCHAIN="stable"'
 [tool.cibuildwheel.linux]
 before-all = "yum install -y wget && {package}/tools/install_rust.sh"
 environment = 'PATH="$PATH:$HOME/.cargo/bin" CARGO_NET_GIT_FETCH_WITH_CLI="true" RUSTUP_TOOLCHAIN="stable"'
+
+[tool.pylint.main]
+extension-pkg-allow-list = ["retworkx", "numpy", "tweedledum", "qiskit._accelerate", "rustworkx"]
+load-plugins = ["pylint.extensions.docparams", "pylint.extensions.docstyle"]
+py-version = "3.7"  # update it when bumping minimum supported python version
+
+[tool.pylint.basic]
+argument-rgx = "[a-z_][a-z0-9_]{2,30}|ax|dt$"
```

Comment:
We're missing some like `attribute-rgx`, which we used to explicitly set whereas the default is now `None`. Pylint says this in its docs:

> Overrides attr-naming-style. If left empty, attribute names will be checked with the set naming style.

But I believe that means it's basically the same as our old defaults? Here, you only set `argument-rgx`, `method-rgx`, and `variable-rgx` because we've changed them from the defaults, right?

Reply:
The new defaults are not given explicitly but they live in https://github.com/PyCQA/pylint/blob/main/pylint/checkers/base/name_checker/naming_style.py

My impression was that we only tweaked the regexes when the defaults felt too strict, so I only carried over the customizations that were needed to pass pylint.

In the format of:
```
regex from old pylintrc
current default implementation of default style
```
here are the different regexes
### The ones where fail pylint without carrying over the customization
#### argument-rgx (snake_case: DEFAULT_NAME_RGX)
```
"[a-z_][a-z0-9_]{2,30}|ax|dt$"
"([^\W\dA-Z][^\WA-Z]{2,}|_[^\WA-Z]*|__[^\WA-Z\d_][^\WA-Z]+__)$"
```
Seems like if we would add ax, dt to `good_names` the default would be better since it has better support for ignored arguments (starting with `_`).

#### method-rgx (snake_case: DEFAULT_NAME_RGX)
```
"(([a-z_][a-z0-9_]{2,49})|(assert[A-Z][a-zA-Z0-9]{2,43})|(test_[_a-zA-Z0-9]{2,}))$"
"([^\W\dA-Z][^\WA-Z]{2,}|_[^\WA-Z]*|__[^\WA-Z\d_][^\WA-Z]+__)$"
```
The testing methods seem worth keeping.

#### variable-rgx (snake_case: DEFAULT_NAME_RGX)
```
"[a-z_][a-z0-9_]{1,30}$"
"([^\W\dA-Z][^\WA-Z]{2,}|_[^\WA-Z]*|__[^\WA-Z\d_][^\WA-Z]+__)$"
```
The main difference is we allow 2-character variables.

### The ones where the default regexes pass our codebase
#### attr-rgx (snake_case: DEFAULT_NAME_RGX)
```
"[a-z_][a-z0-9_]{2,30}$"
"([^\W\dA-Z][^\WA-Z]{2,}|_[^\WA-Z]*|__[^\WA-Z\d_][^\WA-Z]+__)$"
```
Ours was more lenient allowing 2-character vs 3-character in the default. Default seems better.

#### class-attribute-rgx (any)
```
"([A-Za-z_][A-Za-z0-9_]{2,30}|(__.*__))$"
".*"
```
Ours requried minimum 3 characters. Default seems fine.

#### class-rgx (PascalCase)
```
"[A-Z_][a-zA-Z0-9]+$"
"[^\W\da-z][^\W_]+$"
```
Seem equivalent keep the default.

#### const-rgx (UPPER_CASE)
```
"(([A-Z_][A-Z0-9_]*)|(__.*__))$"
"([^\W\da-z][^\Wa-z]*|__.*__)$"
```
Seem equivalent keep the default.

#### function-rgx (snake_case: DEFAULT_NAME_RGX)
```
"[a-z_][a-z0-9_]{2,30}$"
"([^\W\dA-Z][^\WA-Z]{2,}|_[^\WA-Z]*|__[^\WA-Z\d_][^\WA-Z]+__)$"
```
More or less the same keep the default.

#### inlinevar-rgx (any)
```
"[A-Za-z_][A-Za-z0-9_]*$"
".*"
```
Seem equivalent keep the default.

#### module-rgx (snake_case)
```
"(([a-z_][a-z0-9_]*)|([A-Z][a-zA-Z0-9]+))$"
"[^\W\dA-Z][^\WA-Z]*$"
```
Ours was more lenient but we never use PascalCase module names.

Reply:
That works for me. I'm all in favor of minimizing customization when possible. Thanks!

Reply:
> #### argument-rgx (snake_case: DEFAULT_NAME_RGX)
> 
> ```
> "[a-z_][a-z0-9_]{2,30}|ax|dt$"
> "([^\W\dA-Z][^\WA-Z]{2,}|_[^\WA-Z]*|__[^\WA-Z\d_][^\WA-Z]+__)$"
> ```
> 
> Seems like if we would add ax, dt to `good_names` the default would be better since it has better support for ignored arguments (starting with `_`).

This works, and it turns out old regex had a parenthesization error allowing capitalization through.